static/app/components/codeSnippet.tsx

```diff
@@ -38,28 +75,48 @@ export function CodeSnippet({
   className,
   onSelectAndCopy,
   disableUserSelection,
+  tokenReplacers,
 }: CodeSnippetProps) {
   const ref = useRef<HTMLModElement | null>(null);
+  const [tokenNodes, setTokenNodes] = useState<Record<string, HTMLSpanElement[]>>({});
 
   useEffect(() => {
     const element = ref.current;
     if (!element) {
-      return;
+      return () => {};
     }
 
-    if (language in Prism.languages) {
-      Prism.highlightElement(element);
-      return;
+    const applyTokenReplacers = () => {
+      if (!tokenReplacers) {
+        return;
+      }
+      const nodes = replaceTokensWithSpan(element, Object.keys(tokenReplacers));
+      if (Object.keys(nodes).length > 0) {
+        setTokenNodes(nodes);
+      }
+    };
+
+    // Don't load languages in test environment as it takes long enough to trigger
+    // console error about updating state without act() when setState is called afterwords
+    if (language in Prism.languages || NODE_ENV === 'test') {
+      Prism.highlightElement(element, false, applyTokenReplacers);
+      return () => {};
```

Comment:
@vuluongj20 the tests were complaining that the delayed state update in `applyTokenReplaces` was not wrapped in `act`. Maybe you know a better solution for this?
I wanted to avoid to update all tests where `CodeSnippet` occurs somewhere down the component tree 🤔